src/utils/utils.ts

```diff
@@ -1,13 +1,56 @@
 import featuredItemsData from "../data/featureditems.json";
+let featuredItemArray = [];
+
+/**
+ * Sets global featuredItemArray so other functions can use it.
+ **/
+export const setFeaturedItemArray = () => {
+  if (featuredItemArray.length !== 0) {
+    // The global array already has what it needs.
+    return;
+  }
+
+  // If it's empty, now is the time to generate and update the array.
+  let itemsArray = [];
+  for (let i = 0; i < featuredItemsData.featuredItems.images.length; i++) {
+    itemsArray.push(featuredItemsData.featuredItems.images[i].split(".")[0]);
+  }
+  // set featuredItemArray
+  featuredItemArray = itemsArray;
+
+  return featuredItemArray;
+};
 
 /**
  * Returns a random image ID from the list of featured items.
  */
-export const featuredImageID = () => {
-  const randomIndex = Math.floor(
-    Math.random() * featuredItemsData.featuredItems.images.length
-  );
-  return featuredItemsData.featuredItems.images[randomIndex].split(".")[0];
+export const generateRandomImageID = () => {
+  console.log("generating random imageID");
+  if (featuredItemArray.length === 0) {
```

Comment:
removing these lines breaks the app because these two lines are where that method setFeaturedItemArray() is invoked. is there another place to invoke setFeaturedItemArray(), like at the end of the utils file or something?

Reply:
Mentioned offline: let's leave this the way it is for now.